src/lib/challenges/codeFinders.ts

```diff
@@ -20,3 +20,8 @@ export const findAllIconComponents = (code: string) => {
 export const findAllTextInputs = (code: string) => {
   return code.match(/<TextInput[\s\S]*?\/>/gm) ?? [];
 };
+
+export const findTouchableOpacityOpeningTag = (code: string) => {
+  const touchableOpacityTag = code.match(/<TouchableOpacity[\s\S]*?>/gm)
+  return touchableOpacityTag ? touchableOpacityTag[0] : ''
```

Comment:
I think we should return undefined here instead - this more clearly means "we did not find a touchable opacity", compared to using an empty string to mean something else, which could be confusing further down the line when using the result of this. If we return undefined, calls to this will rightfully be warned to handle the possibility of there being no touchable opacity, compared to failing quietly and letting us do something weird with the empty string

Reply:
Good point - updating to returned undefined and explicitly handling this case where the function is used